applications/StructuralMechanicsApplication/custom_constitutive/linear_isotropic_damage_3D_law.h

```diff
@@ -0,0 +1,334 @@
+     * @see Parameters
+     */
+    void FinalizeMaterialResponsePK1(Parameters& rValues) override;
+
+    /**
+     * @brief Finalize the material response in terms of 2nd Piola-Kirchhoff stresses
+     * @param rValues The specific parameters of the current constitutive law
+     * @see Parameters
+     */
+    void FinalizeMaterialResponsePK2(Parameters& rValues) override;
+
+    /**
+     * @brief Finalize the material response in terms of Kirchhoff stresses
+     * @param rValues The specific parameters of the current constitutive law
+     * @see Parameters
+     */
+    void FinalizeMaterialResponseKirchhoff(Parameters& rValues) override;
+
+    /**
+     * @brief Finalize the material response in terms of Cauchy stresses
+     * @param rValues The specific parameters of the current constitutive law
+     * @see Parameters
+     */
+    void FinalizeMaterialResponseCauchy(Parameters& rValues) override;
+
+    /**
+     * @brief calculates the value of a specified variable
+     * @param rParameterValues the needed parameters for the CL calculation
+     * @param rThisVariable the variable to be returned
+     * @param rValue a reference to the returned value
+     * @return rValue output: the value of the specified variable
+     */
+
+    /**
+     * @brief calculates the value of a specified variable
+     * @param rParameterValues the needed parameters for the CL calculation
+     * @param rThisVariable the variable to be returned
+     * @param rValue a reference to the returned value
+     * @return rValue output: the value of the specified variable
+     */
+    double& CalculateValue(Parameters& rParameterValues,
+                           const Variable<double>& rThisVariable,
+                           double& rValue) override;
+
+    /**
+     * @brief This function provides the place to perform checks on the completeness of the input.
+     * @details It is designed to be called only once (or anyway, not often) typically at the beginning
+     * of the calculations, so to verify that nothing is missing from the input or that no common error is found.
+     * @param rMaterialProperties The properties of the material
+     * @param rElementGeometry The geometry of the element
+     * @param rCurrentProcessInfo The current process info instance
+     */
+    int Check(
+        const Properties& rMaterialProperties,
+        const GeometryType& rElementGeometry,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override {
+        rOStream << "Linear Isotropic Damage 3D constitutive law\n";
+    };
+
+protected:
+
+    ///@name Protected static Member Variables
+    ///@{
+    ///@}
+
+    ///@name Protected member Variables
+    ///@{
+    bool mInelasticFlag; /// Flags when in inelastic regime
+    double r; /// Damage threshold
```

Comment:
can you please you a more descriptive name?
also it should be preceeded by `m`

Reply:
True. Done.